server/models/index.js

```diff
@@ -35,42 +35,70 @@ const addTags = async (tags, titleOfSong) => {
 
 const getAllSongsHome = async () => {
   db = process.env.NODE_ENV === 'test' ? global.client : db;
-  const result = await db.query(`SELECT * FROM ${songsTable}`);
-  for (const song of result.rows) {
-    // console.log("🚀 ~ file: index.js:38 ~ getAllSongsHome ~ song:", song)
-    const findUsername = await db.query(`SELECT username FROM ${usersTable} WHERE id=$1`, [song.user_id]);
-    song.username = findUsername.rows[0].username;
-    // console.log("🚀🚀 ~ file: index.js:40 ~ getAllSongsHome ~ findUsername:", findUsername)
-  }
+  // const result = await db.query(`SELECT * FROM ${songsTable}`);
+  // const result = await db.query(`SELECT songs.*, users.username, ARRAY_AGG(song_tags.name) AS tags FROM songs INNER JOIN users ON songs.user_id = users.id INNER JOIN song_tags ON songs.id = song_tags.song_id GROUP BY songs.id, users.id`);
+  const result = await db.query(`SELECT
+    songs.*,
+    users.username,
+    json_object_agg(
+      song_tags.name,
+      json_build_object(
+        'id', song_tags.id,
+        'song_id', song_tags.song_id,
+        'name', song_tags.name
+      )
+    ) AS tags
+  FROM
+    songs
+    INNER JOIN users ON songs.user_id = users.id
+    INNER JOIN song_tags ON songs.id = song_tags.song_id
+  GROUP BY
+    songs.id,
+    users.id`)
+    .catch(err => console.log(`error retrieving songs on home tab`, err));
+
   return result.rows;
 };
 
 
-const getAllSongs = async (user) => {
+const getSongsByUser = async (user) => {
   db = process.env.NODE_ENV === 'test' ? global.client : db;
   const userId = await db.query(`SELECT id FROM ${usersTable} WHERE name = '${user}'`);
+  // const userId = await db.query(`SELECT id FROM ${usersTable} WHERE username = '${user}'`);
   if (!userId.rows.length) return [];
-  const result = await db.query(`SELECT json_agg(
-    json_build_object(
-      'id', id,
-      'title', title,
-      'created_at', created_at,
-      'path_to_song', path_to_song,
-      'play_count', play_count,
-      'fav_count', fav_count,
-      'path_to_artwork', path_to_artwork,
-      'user_id', user_id,
-      'tags', (
-        SELECT coalesce (json_agg(
+  const result = await db.query(`SELECT
+        json_agg(
           json_build_object(
-            'id', id,
-            'name', name,
-            'song_id', song_id
+            'id', ${songsTable}.id,
+            'title', ${songsTable}.title,
+            'created_at', ${songsTable}.created_at,
+            'path_to_song', ${songsTable}.path_to_song,
+            'play_count', ${songsTable}.play_count,
+            'fav_count', ${songsTable}.fav_count,
+            'path_to_artwork', ${songsTable}.path_to_artwork,
+            'user_id', ${songsTable}.user_id,
+            'username', ${usersTable}.username,
+            'tags', (
+              SELECT coalesce (
+                json_agg(
+                  json_build_object(
```

Comment:
We probably just need the tags.name value here, song_id is already present in the id field, and I can't see when we would need tags.id